myapp/app/models/user.rb

```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  has_secure_password
+  validates :username, length: { in: 3..20 }
+  validates :password, presence: true, confirmation: true, format: { with: /\A[^\s]*\z/ }, length: { in: 5..15 }
+
   has_many :tasks, dependent: :destroy
-  validates :username, format: { with: /\A[^\s]*\z/, message: 'should not contain space!' }
+  validates :username, format: { with: /\A[^\s]*\z/ }
```

Comment:
I am not sure why you have removed the validation message for space on `username`.
And checking space on `password`, since space can contain is password. right?

do you want to keep your application rules like this? if its not a mistake then its okay to be unusual. let me know. 

Rest of them are LGTM

Reply:
I just removed the error message since it's defined in i18n's yml file. The validation for space is still remaining.

Reply:
I see, so you want to restrict space in password too?

Reply:
Yes. I remember you just recommended me to add space restriction into password last week.

Reply:
Ah! sorry for the confusion! i mean to say about `username` only. but password generally can have also space. 
anyways if you want to change you can later or its okay, either its fine for this training. 

LGTM 👍